omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_checks.rb

```diff
@@ -0,0 +1,54 @@
+class PreflightValidationFailed < Exception
+
+end
+
+class PreflightValidator
+  attr_reader :node, :previous_run, :helper
+  def initialize(node)
+    @helper = OmnibusHelper.new(node)
+    @node = node
+    @previous_run = node['previous_run']
+    def fail_with error
+      raise PreflightValidationFailed, error
+    end
+  end
+
+  def backend?
+    #
+    # When these preflight checks are run, the chef-server.rb has been ingested into
+    # PrivateChef but has not been merged into the node. This means for accurate results, we'll
+    # need to provide an assembled node object that contains the configuration that
+    # the backend check needs.
+    faux_node = { 'enterprise' => node['enterprise'],
+                  node['enterprise']['name'] =>  PrivateChef }
+    return unless EnterpriseChef::Helpers.backend? faux_node
+
+    return PrivateChef['topology'] == 'standalone' || PrivateChef['role'] == 'backend'
```

Comment:
The logic here is a bit odd to me.  Can we replace both of these `return` lines with the following single line:

```
EnterpriseChef::Helpers.backend? faux_node
```

or am I missing some nuance here?